csi.proto

```diff
@@ -576,8 +576,9 @@ message ControllerPublishVolumeRequest {
   // This is a REQUIRED field.
   VolumeCapability volume_capability = 3;
 
-  // Whether to publish the volume in readonly mode. This field is
-  // REQUIRED.
+  // Whether to publish the volume in readonly mode. This field MUST be
+  // specified only if SP has PUBLISH_READONLY controller capability.
+  // This is a REQUIRED field.
   bool readonly = 4;
```

Comment:
Is making `readonly` flag optional simpler?

Reply:
Problem with making it optional is that the CO doesn't know when to set it. If CO sets it to true and a plugin doesn't support it, what happens? Consider the case where the CO uses the same flag for both ControllerPublish and NodePublish readonly -- we don't want to get in to a case where the CO thinks a volume is readonly but it's not.